tyssue/core/history.py

```diff
@@ -158,7 +176,113 @@ def __iter__(self):
 
+class HistoryHdf5(History):
+    """ This class handles recording and retrieving time series
+    of sheet objects.
+
+    """
+
+    def __init__(self, sheet, save_every=0, dt=0, extra_cols=None, path="", overwrite=False):
+        """Creates a `SheetHistory` instance.
+
+        Parameters
+        ----------
+        sheet : a :class:`Sheet` object which we want to record
+        extra_cols : dictionnary with sheet.datasets as keys and list of
+            columns as values. Default None
+        """
+        if path is None:
+            warnings.warn(
+                "No directory is given. The HDF5 file will be saved in the working directory.")
+            self.path = os.os.getcwd()
+        else:
+            self.path = path
+
+        if os.path.exists(os.path.join(self.path, 'out.hf5')):
+            if overwrite:
+                self.hf5file = os.path.join(self.path, 'out.hf5')
+                pd.HDFStore(self.hf5file, 'w').close()
+                warnings.warn(
+                    "The file already exist and will be overwritten.")
+            else:
+                expend = 1
+                while True:
+                    expend += 1
+                    new_file_name = "out{}.hf5".format(str(expend))
+                    if os.path.exists(os.path.join(self.path, new_file_name)):
+                        continue
+                    else:
+                        self.hf5file = os.path.join(self.path, new_file_name)
+                        pd.HDFStore(self.hf5file, 'w').close()
+                        warnings.warn(
+                            "The file already exist and the new filename is {}".format(new_file_name))
+                        break
+
+        else:
+            self.hf5file = os.path.join(self.path, 'out.hf5')
+            pd.HDFStore(self.hf5file, 'w').close()
+        History.__init__(self, sheet, save_every, dt, extra_cols)
+
+    def record(self, to_record=None, time_stamp=None):
+        """Appends a copy of the sheet datasets to the history instance.
+
+        Parameters
+        ----------
+        to_report : list of strings, default ['vert']
+            the datasets from self.sheet to be saved
+
+        """
+        if to_record is None:
+            to_record = ["vert"]
+
+        if time_stamp is not None:
+            self.time = time_stamp
+        else:
+            self.time += 1
+
+        if self.index % (int(self.save_every / self.dt)) == 0:
+            for element in to_record:
+                df = self.sheet.datasets[element]
+                times = pd.Series(
+                    np.ones((df.shape[0],)) * self.time, name="time")
+                df = pd.concat([df, times], ignore_index=False,
+                               axis=1, sort=False)
+
+                with pd.HDFStore(self.hf5file, 'a') as file:
+                    try:
+                        file.append(key="{}_df".format(element),
+                                        value=df)
+                    except:
```

Comment:
I did that because `append` work only if the table already exist. But I just see that `put` method as an append arg to complete an existing table. I thought that `put` just erase the table if it is already exist. 
I simplify...

Reply:
well, it didn't work... The table need to be created before append someting... It is logical...
Do you think it is better to test explicitly the presence of vert_df instead of using try/except ?

Reply:
I don't know which is better. Just have the except be specific to the error raised (`KeyError`?)

Reply:
Depsite the doc says, I can use the append method alone. Apparently it seems to create the table if it not exist...

Reply:
cool (maybe - if you feel like it) - you can open an Issue on pandas website and fix the doc ;)